master/internal/api_user.go

```diff
@@ -683,3 +683,116 @@ func (a *apiServer) PostUserActivity(
 	}
 	return &apiv1.PostUserActivityResponse{}, err
 }
+
+func (a *apiServer) PostLongLivedToken(
+	ctx context.Context, req *apiv1.PostLongLivedTokenRequest,
+) (*apiv1.PostLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if err = user.AuthZProvider.Get().CanCreateUsersOwnToken(
```

Comment:
nit (non-blocking): Definitely take this with a grain of salt, but could this name be confusing? It might be better to add a descriptor to the Token part of this function name so it's clearer what we're checking. Maybe something like `CanCreateUsersOwnAccessToken` & we could try to match the `CanAdministrateUser` function for `CanCreateUsersToken` so that it instead would be `CanAdministrateAccessToken`.

By all means these are suggestions, and if I can absolutely be convinced that `CanCreateUsersOwnToken` and `CanCreateUsersToken` are the clearest ways to communicate these checks

Reply:
I was thinking the same, but was thinking to change the names to `CanCRUDUserOwnAccessToken` and `CanCRUDUserAccessToken`. This will convey the description better. Your thoughts?

Reply:
I would be more inclined to separate out the individual operation into there own permissions/checks, that way in the future if we ever expose RBAC permissions to allow folks to create custom roles they can have more granular control.

Reply:
Agreed!

Reply:
Resolving this comment as we have a consensus on this ask and have pushed those changes.